base/shared/src/main/scala/scalaz/data/io.scala

```diff
@@ -0,0 +1,83 @@
+package scalaz
+package data
+
+import scalaz.zio.Errors.NothingRaced
+import scalaz.zio.{ Fiber, IO }
+import scalaz.tc.Foldable
+
+import scala.{ AnyVal, Either, Nothing, Option, Unit }
+import scala.collection.generic.CanBuildFrom
+import scala.collection.mutable
+import scala.inline
+
+object io {
+
+  private final def _bracket0[E, A, B](
+    acquire: IO[E, A]
+  )(release: (A, Maybe[E \/ B]) => IO[Nothing, Unit])(use: A => IO[E, B]): IO[E, B] = {
+    val release1: (A, Option[Either[E, B]]) => IO[Nothing, Unit] =
+      (a: A, o: Option[Either[E, B]]) => release(a, Maybe.fromOption(o.map(Disjunction.fromEither[E, B])))
+    IO.bracket0[E, A, B](acquire)(release1)(use)
+  }
+
+  implicit class IOObjOps(private val ioObj: IO.type) extends AnyVal {
+
+    final def absolve[E, A](v: IO[E, E \/ A]): IO[E, A] =
```

Comment:
Why is `absolve` in the companion instead of in the `IO` type? That's something I always wondered in zio too. It always breaks dot syntax for me. Did you have a reason @jdegoes ?

Reply:
We would need `ev` magic to do it into `IO`. I guess I'm fine with that. In the past I've stayed away.